crates/cheatnet/src/forking/data.rs

```diff
@@ -0,0 +1,38 @@
+use crate::runtime_extensions::forge_runtime_extension::contracts_data::build_name_selector_map;
+use starknet::core::types::ContractClass;
+use starknet::core::types::contract::AbiEntry;
+use starknet_api::core::{ClassHash, EntryPointSelector};
+use std::collections::HashMap;
+
+#[derive(Default)]
+pub struct ForkData {
+    pub abi: HashMap<ClassHash, Vec<AbiEntry>>,
+    pub selectors: HashMap<EntryPointSelector, String>,
+}
+
+impl ForkData {
+    /// Creates a new instance of [`ForkData`] from a given `fork_compiled_contract_class`.
+    #[must_use]
+    pub fn new(fork_compiled_contract_class: &HashMap<ClassHash, ContractClass>) -> Self {
+        let abi: HashMap<ClassHash, Vec<AbiEntry>> = fork_compiled_contract_class
+            .iter()
+            .filter_map(|(class_hash, contract_class)| {
+                let ContractClass::Sierra(sierra_class) = contract_class else {
+                    return None;
+                };
```

Comment:
I assume it's impossible to try fetch abis from cairo 0 contracts down the line?